### DISCUSSION TITLE
Multiple local subnet configured for Traffic selector

### DISCUSSION THREAD

**Discussion by muhammadi123** [strongswan/strongswan#1277] [Q&A]
**Case 1:** we have a multiple ipsec tunnels configured on ipv4. If we don't add VIP(Virtual IP) as TS in local subnet configuration in ipsec server then the traffic doesn't allow via tunnel .
 If we add VIP as TS along with the expected subnet as TS(i.e,VIP subnet + expected subnet) then only we find traffic  is going via ipsec tunnel.
Is  it the expected behaviour? 
**Case 2:**  Same setup we have multiple ipsec tunnel configured on ipv6  then we don't need to add here multiple local subnet(i.e., VIP subnet + expected subnet) but still we see traffic is going via ipsec tunnel.

**Comment by Thermi** [comment]
Please provide exact configurations and descriptions of all networks.
Please follow #196

**Reply by muhammadi123** [reply]
Hi Thermi,
Thanks for the quick response .

The Setup is like this 
![image](https://user-images.githubusercontent.com/36219804/188809603-9bef1557-b614-4bb1-ae48-1760e14870f9.png)

**Configurations from ipsec - client** 
 cat /opt/strongswan/etc/ipsec.conf 
config setup
        strictcrlpolicy=no
        charondebug="ike 1, knl 1, cfg 1, net 1, esp 1, dmn 1, mgr 1"

conn %default
        keyexchange=ikev2
        auto=add
        keyingtries=5
        ikelifetime=864000
        keylife=864000
        rekeymargin=540
        rekeyfuzz=0%
        ike=aes256-sha1-modp2048, aes256-sha256-modp2048, aes192-sha1-modp2048, aes192-sha256-modp2048, aes128-sha1-modp2048, aes128-sha256-modp2048, 3des-sha1-modp2048, 3des-sha256-modp2048!
        esp=aes256-sha1, aes256-sha256, aes192-sha1, aes192-sha256, aes128-sha1, aes128-sha256, 3des-sha1, 3des-sha256!
        dpdaction=clear
        dpddelay=30
        leftsourceip=%config
        reauth=no

conn S1_1
        mobike=no
        left=172.18.50.115
        leftid=00A00ACDF9DA.airspan.com
        leftcert=00A00ACDF9DA.cer
        right=172.18.53.10
        rightid=*
        rightca="C=US, O=Airspan Networks Inc., CN=Airspan Networks Inc. 2048-bit Root CA"
        rightikeport=500
        rightsubnet=0.0.0.0/0
        type=tunnel
        auto=add
        leftupdown=/bs/bin/updownScript.sh
conn S1_2
        mobike=no
        left=172.18.50.115
        leftid=00A00ACDF9DA.airspan.com
        leftcert=00A00ACDF9DA.cer
        right=172.18.54.253
        rightid=*
        rightca="C=US, O=Airspan Networks Inc., CN=Airspan Networks Inc. 2048-bit Root CA"
        rightikeport=500
        rightsubnet=%dynamic
		leftsubnet =0.0.0.0/0
        type=tunnel
        auto=add
        leftupdown=/bs/bin/updownScript.sh
        
**Server configuration from ACP & from FB**
NMS/ACP IPSec Config :-
        [rightid=@00A00A004A86.airspan.com](mailto:rightid=@00A00A004A86.airspan.com)
        fragmentation=yes
        leftsubnet=172.18.51.0/24,172.18.2.0/24
        rightsourceip=172.18.51.58
        rightdns=172.18.50.248
FB IPsec Config :-
        [rightid=@00A00A004A86.airspan.com](mailto:rightid=@00A00A004A86.airspan.com)
        leftsubnet=172.18.41.0/24,172.17.156.0/24
        rightsourceip=172.18.41.22
        rightdns=172.18.50.248
        fragmentation=yes

Thanks,
Muhammadi.

**Reply by Thermi** [reply]
Hi, when you're using such a broad traffic selector like 0.0.0.0/0 == 0.0.0.0/0 you are tunneling ANY traffic, regardless of the source. That is probably way too much. Omitting leftsubnet will help you here. It dynamically defaults to the correct local address automatically.

**Reply by muhammadi123** [reply]
Hi Thermi,
As we are handling with multiple tunnels so we can't go with TS as ANY like 0.0.0.0/0 .
Here as per the above setup we have 3 tunnels with Netspan, MOCN and EPC.

The param "leftsubnet" defined below in the following ipsec servers MOCN GW , Netspan GW and EPC GW.
-> MOCN is using the default traffic do not require TS
     leftsubnet=0.0.0.0/0
-> Netspan is using TS 
     leftsubnet=172.18.51.0/24,172.18.2.0/24
     Note: 172.18.51.0/24 is VIP subnet
-> EPC is using TS 
     leftsubnet=172.18.41.0/24,172.17.156.0/24
     Note: 172.18.41.0/24 is VIP subnet
 
 But only in ipv4 case we have to define VIP subnet as well then only the traffic is going via tunnel else it is not working
